api/v1alpha1/clusterconfig_types.go

```diff
@@ -266,6 +269,18 @@ type User struct {
 	Sudo string `json:"sudo,omitempty"`
 }
 
+// Encryption defines the configuration to enable encryption at REST
+// This configuration is used by API server to encrypt data before storing it in ETCD.
+// Currently the encryption only enabled for secrets and configmaps.
+type Encryption struct {
+	// Encryption providers
+	// +kubebuilder:validation:UniqueItems=true
+	// +kubebuilder:validation:Enum=aescbc;aesgcm
+	// +kubebuilder:default=aescbc
+	// +optional
+	Providers []string `json:"providers"`
```

Comment:
The API supports multiple providers. https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/#understanding-the-encryption-at-rest-configuration
should we support multiple providers or single provider? 

The first provider in the list will be used for encryption.

Reply:
I think the only reason for multiple providers here is if a user wants to change providers, they will be tried in order for decryption, the first provider being used for encryption on write. I think that use case is quite advanced/unusual so I would advocate for only a single provider at this point. We can extend the API later if we need to support multiple. Wdyt?

Reply:
Agree. We can keep it limited to one provider. However I think we should keep type as array and add max length to 1 to allow adding more providers in future.

Reply:
Ok that sounds good to me.